File: public/PublicMedia/static/solutions/python/problem1.py
""" 
Purpose: Find the sum of all the multiples of 3 or 5 below 1000.

Author: Eben Quenneville
Date: 2022-09-03, modified 2023-01-29
"""
max = 1000

def problem1() -> int:
    sum = 0
    for i in range(1, max):
        if i % 3 == 0:
            sum += i
            continue
        if i % 5 == 0:
            sum += i
    return sum


def main():
    solution = problem1()
    assert solution == 233168
    print(f"Output: {solution}")


if __name__ == "__main__":
    main()

File: public/PublicMedia/static/solutions/zig/problem1.zig
// Purpose: Find the sum of all the multiples of 3 or 5 below 1000.
// Author: Eben Quenneville
// Date: 2022-11-12
const std = @import("std");
const assert = std.debug.assert;

pub fn main() !void {
    var sum: usize = 0;
    var i: usize = 0;
    while (i < 1000) : (i += 1) {
        if (@mod(i, 3) == 0) {
            sum += i;
            continue;
        }
        if (@mod(i, 5) == 0) {
            sum += i;
        }
    }
    const stdout = std.io.getStdOut().writer();
    assert(sum == 233168);
    try stdout.print("Output: {d}\n", .{sum});
}
